.azure-pipelines.yml

```diff
@@ -57,7 +57,7 @@ jobs:
           set -e
           uv pip install --system --compile tomli packaging
           deps=`python3 ci/scripts/min-deps.py pyproject.toml --extra dev test`
-          uv pip install --system --compile $deps pytest-cov "anndata @ ."
+          uv pip install --system --compile $deps pytest-cov "anndata[test,dev] @ ."
```

Comment:
@flying-sheep I think this is a bug, no? The min-deps are created without respect for upper bounds so the constraints can end up not including the upper bounds because the min-deps script doesn't care about it? I think this is the correct fix given the fact that the script is called "min-deps" and not "min-deps-with-upper-bound" but maybe I'm off

Reply:
Good point! However I think the better fix would be to change the script so it leaves upper bounds (and things like environment markers) alone.

The script does already do more than just minimal deps, as it only pins deps with minimum version boundaries. Deps without that (only upper bounds, no bounds, …) are passed through without any minimizing!

So I’d say we could do two things:
1. change the script to either leave upper bounds alone or just skip deps without a minimum boundary entirely (just returning pinned dependencies)
2. depending on the decision for 1., the workflow would stay unchanged from `main` or we should change the CLI so it takes a comma separated list of extras and reuse it for both locations (`--extra=$extras` and `"anndata[$extras] @ ."`)